### ISSUE TITLE
ElggEntity should we prevent changing type/subtype by the magic setter

### ISSUE THREAD

**Issue by jeabakker** [Elgg/Elgg#13289]
Thus only allow it to be changed by some helper function

```php
// not allowed
$entity->type = 'object';
$entity->subtype = 'subtype';

// allowed
$entity->setType('object');
$entity->setSubtype('subtype');
```

**Comment by sarathsince85** [comment]
What about changing the container_guid too? Will be useful in some cases..

**Comment by jeabakker** [comment]
> I think should not be prevented. Let it be more flexible.

but it also can introduce unintended issues

> We should allow changing container_guid also. Will be useful in some cases..

this is allowed

**Comment by jeabakker** [comment]
fixed in #13401
